xdsl/dialects/seq.py

```diff
@@ -0,0 +1,87 @@
+"""
+CIRCT’s seq dialect
+
+[1] https://circt.llvm.org/docs/Dialects/Seq/
+"""
+
+from xdsl.dialects.builtin import (
+    AnyIntegerAttr,
+    IntegerAttr,
+    IntegerType,
+    Signedness,
+    TypeAttribute,
+)
+from xdsl.ir import Dialect, OpResult
+from xdsl.irdl import (
+    IRDLOperation,
+    Operand,
+    ParametrizedAttribute,
+    attr_def,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    result_def,
+)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import VerifyException
+
+u64 = IntegerType(64, Signedness.UNSIGNED)
+u32 = IntegerType(32, Signedness.UNSIGNED)
+
+
+@irdl_attr_definition
+class ClockType(ParametrizedAttribute, TypeAttribute):
+    """
+    A type for clock-carrying wires. Signals which can be used to drive the clock input of sequential operations.
+    """
+
+    name = "seq.clock"
+
+
+clock = ClockType()
+
+
+@irdl_op_definition
+class ClockDivider(IRDLOperation):
+    """Produces a clock divided by a power of two"""
+
+    name = "seq.clock_div"
+
+    pow2: IntegerAttr = attr_def(AnyIntegerAttr)
```

Comment:
```suggestion
    pow2 = attr_def(AnyIntegerAttr)
```
You can just drop this